### PR TITLE
Sequential Markowitz model + doc

### PR DESCRIPTION
**Pull request by bqth29**
## PR decription

Following the discussion in #17, the Markowitz API has been enhanced with a new sequential model that allow an optimization of the portfolio relying among others on rebalancing costs which couple the timestamps together:

$$\hbox{argmax } \sum_{t = 1}^{T} w_{t}^{T} \mu_{t} - \frac{\gamma}{2} w_{t}^{T} \Sigma_{t} w_{t} - \Delta w_t^T \Lambda_t \Delta w_t$$

This model is implemented as the `SequentialMarkowitz` class of the `simulated_bifurcation.models` module.

## New features

- The `SequentialMarkowitz` model has been added to the models pool (`simulated_bifurcation.models` module)
- It is now possible to create optimization models without explicitly importing the  `simulated_bifurcation.models` module

```python
import simulated_bifurcation as sb

qubo = sb.models.QUBO()
```

- Minor fixes on the `Markowitz` class
- Unit tests have been added to cover the optimization models
- Other minor fixes
- Improved docstrings

### CODE REVIEW

File: src/simulated_bifurcation/models/utils.py
Comment (BusyBeaver-42) [inline src/simulated_bifurcation/models/utils.py:1]: Duplication of `IsingPolynomialInterface.__cast_matrix_to_tensor`

File: src/simulated_bifurcation/models/markowitz.py [hunk 1]
```diff
@@ -24,23 +25,121 @@ def __init__(
...
+        number_of_bits: int = 1,
+        dtype: torch.dtype = torch.float32,
+        device: str = "cpu",
+    ) -> None:
+        """
+        covariances : T x N x N
+        expected_returns : T x N
+        rebalancing_costs : T x N x N
+        """
+        self.covariances = cast_to_tensor(covariances)
```
Comment (BusyBeaver-42) [inline src/simulated_bifurcation/models/markowitz.py:74]:
```suggestion
        self.covariances = self.__cast_to_tensor(covariances, dtype, device)
```

File: src/simulated_bifurcation/models/markowitz.py [hunk 2]
```diff
@@ -24,23 +25,121 @@ def __init__(
...
+        self.expected_returns = cast_to_tensor(expected_returns)
```
Comment (BusyBeaver-42) [inline src/simulated_bifurcation/models/markowitz.py:75]:
```suggestion
        self.expected_returns = self.__cast_to_tensor(expected_returns, dtype, device)
```

File: src/simulated_bifurcation/models/markowitz.py [hunk 3]
```diff
@@ -24,23 +25,121 @@ def __init__(
...
+        self.rebalancing_costs = cast_to_tensor(rebalancing_costs)
```
Comment (BusyBeaver-42) [inline src/simulated_bifurcation/models/markowitz.py:76]:
```suggestion
        self.rebalancing_costs = self.__cast_to_tensor(rebalancing_costs, dtype, device)
```

File: src/simulated_bifurcation/models/markowitz.py [hunk 4]
```diff
@@ -1,9 +1,10 @@
-from typing import Optional, Union
+from typing import Optional, Tuple, Union
 
 import numpy as np
 import torch
 
 from ..polynomial import IntegerPolynomial
+from .utils import cast_to_tensor
```
Comment (BusyBeaver-42) [inline src/simulated_bifurcation/models/markowitz.py:7]:
```suggestion
```

File: src/simulated_bifurcation/models/markowitz.py
Comment (BusyBeaver-42) [inline src/simulated_bifurcation/models/markowitz.py:1]: # Code duplication:
- `portfolio` property defined both in `Markowitz` and `SequentialMarkowitz` with the same code
- `gains` property defined both in `Markowitz` and `SequentialMarkowitz` with the same code
- `gains` emulates the behaviour of the `best_only` argument when calling an optimization method

# Inconsistencies
- `gains` returns `0` when no optimization method has been called which is inconsistent whith the behaviour of the package (should return `None`)

**Review by BusyBeaver-42 — CHANGES_REQUESTED** [review]

File: src/simulated_bifurcation/models/markowitz.py
Comment (bqth29) [reply src/simulated_bifurcation/models/markowitz.py:1]: The behavior has been optimized by making `Markowitz` inherit from `SequentialMarkowitz`

**Pull request merged**